arch/riscv/core/isr.S

```diff
@@ -342,9 +746,11 @@ on_thread_stack:
 	addi t2, t2, -1
 	sw t2, _kernel_offset_to_nested(t1)
 
+#if (CONFIG_USERSPACE == 0) && (CONFIG_PMP_STACK_GUARD == 0)
```

Comment:
do not compare to 0 value, that is not how Kconfig works
use #ifdef or #if defined(...)

Reply:
Fixed